torch/_higher_order_ops/auto_functionalize.py

```diff
@@ -365,7 +391,24 @@ def __call__(
 auto_functionalized_v2.fallthrough(DispatchKey.AutogradCUDA)
 
 
-def can_auto_functionalize(op: OperatorBase) -> bool:
+def can_auto_functionalize(
+    op: Union[OperatorBase, HigherOrderOperator],
+    hop_schema: Optional[torch.FunctionSchema] = None,
+) -> bool:
```

Comment:
So I don't like this very much because it muddies the API. The `can_auto_functionalize` API was pretty clear before.

I wonder if it would help to have a helper "HOPInstance" class. A HOPInstance is a HOP and the schema of the HOP. Before calling into can_auto_functionalize, we create a HOPInstance and then pass it to can_auto_functionalize.

Feel free to push back.

Reply:
I like the idea. To clarify, this is only for calling into can_auto_functionalize, right? Potentially, we could extend to dispatch key implementations and other hops.